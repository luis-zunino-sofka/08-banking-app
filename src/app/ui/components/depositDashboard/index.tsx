import { useDeposit } from "@core/hooks";
import { decryptAES } from "@core/utils";
import { Toaster } from "react-hot-toast";
import { Button } from "../button";

export const DepositDashboard = () => {
  const {
    isLoading,
    accounts,
    errors,
    isAccountSelected,
    handleSubmit,
    register,
    onSubmit,
  } = useDeposit();

  return (
    <div className="deposit">
      <form
        id="deposit-form"
        className="form"
        onSubmit={handleSubmit(onSubmit)}
      >
        <label htmlFor="deposit-source">
          Selecciona el origen del depósito:
        </label>
        <select
          id="deposit-source"
          {...register("depositSource", {
            required: "El monto a depositar es obligatorio",
          })}
        >
          <option value="branch-deposit" defaultValue="branch-deposit">
            Depósito desde sucursal (Sin costo)
          </option>
          <option value="atm">
            Depósito desde cajero automático ($2 de costo)
          </option>
          <option value="account">
            Depósito desde otra cuenta ($1.5 de costo)
          </option>
        </select>
        {errors.depositSource && (
          <span className="error-message">{errors.depositSource.message}</span>
        )}
        <label htmlFor="deposit-amount">Monto:</label>
        <input
          type="number"
          id="deposit-amount"
          placeholder="Monto a depositar"
          required
          min={0}
          {...register("amount", {
            required: "El monto a depositar es obligatorio",
            min: {
              value: 1,
              message: "El monto a depositar debe ser mayor a 0",
            },
          })}
        />
        {errors.amount && (
          <span className="error-message">{errors.amount.message}</span>
        )}
        <div>
          <label htmlFor="accountId">
            {isAccountSelected
              ? "Cuenta de extracción:"
              : "Cuenta a depositar:"}
          </label>
          <select
            id="accountId"
            {...register("accountId", {
              required: "La cuenta bancaria es obligatorio",
            })}
            required
          >
            <option value="">Elige una cuenta</option>
            {accounts.map((a) => (
              <option key={a.accountId} value={a.accountId}>
                Cuenta: {decryptAES(a.encryptedNumber)} - Monto: {a.amount}
              </option>
            ))}
          </select>

          {errors.accountNumberToDeposit && (
            <span className="error-message">
              {errors.accountNumberToDeposit.message}
            </span>
          )}
        </div>
        {isAccountSelected && (
          <>
            <label htmlFor="account-number-to-deposit">
              Selecciona la cuenta a depositar:
            </label>
            <input
              type="text"
              id="account-number-to-deposit"
              placeholder="Cuenta a depositar"
              required
              min={0}
              {...register("accountNumberToDeposit", {
                required: "La cuenta a depositar es obligatorio",
              })}
            />

            {errors.accountNumberToDeposit && (
              <span className="error-message">
                {errors.accountNumberToDeposit.message}
              </span>
            )}
          </>
        )}
        <Button label="Depositar" isLoading={isLoading} />
        <p>* El costo de la transacción se deducirá del saldo.</p>
      </form>
      <Toaster />
    </div>
  );
};
