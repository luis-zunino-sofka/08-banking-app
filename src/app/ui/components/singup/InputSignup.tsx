/* eslint-disable @typescript-eslint/ban-ts-comment */
import { SignupFormData } from "@core/interfaces";
import { FieldErrors, UseFormRegisterReturn } from "react-hook-form";

interface IInput {
  register: UseFormRegisterReturn<string>;
  name: string;
  type: string;
  label: string;
  errors: FieldErrors<SignupFormData>;
  placeHolder: string;
}

export const InputSignup = (props: IInput) => {
  const { register, name, type, label, errors, placeHolder } = props;
  return (
    <div className="signup-form__input-group">
      <label htmlFor={`input-${name}`} className="signup-form__label">
        {label}
      </label>
      <input
        id={`input-${name}`}
        type={type}
        className="signup-form__input"
        placeholder={placeHolder}
        {...register}
      />
      {/* @ts-expect-error */}
      {errors[name] && (
        /* @ts-expect-error */
        <span className="error-message">{errors[name].message}</span>
      )}
    </div>
  );
};
