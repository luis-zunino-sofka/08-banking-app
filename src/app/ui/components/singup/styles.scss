@use "../../styles/variables" as var;

.signup-form {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #f4f7fa;
  font-family: Arial, sans-serif;

  &__form {
    background-color: var.$bg-color-white;
    height: Calc(100vh - 20px);
    padding: 30px;
    border-radius: 8px;
    box-shadow: 0 4px 20px rgba(0, 0, 0, 0.1);
    width: 100%;
    max-width: fit-content;
    box-sizing: border-box;
  }

  &__title {
    font-size: 28px;
    font-weight: bold;
    margin-bottom: 20px;
    text-align: center;
  }

  &__input-group {
    margin-bottom: 20px;
    display: flex;
    flex-direction: column;
  }

  &__label {
    font-size: 16px;
    font-weight: 600;
    margin-bottom: 8px;
    display: block;
  }

  &__input {
    width: 250px;
    padding: 12px;
    font-size: 16px;
    border: 1px solid #ccc;
    border-radius: 5px;
    box-sizing: border-box;
    transition: border 0.3s ease;

    &:focus {
      border-color: #007bff;
      outline: none;
    }
  }

  &__error {
    color: var.$text-color-error;
    font-size: 12px;
    margin-top: 5px;
  }

  &__submit {
    background-color: #007bff;
    color: white;
    font-size: 16px;
    font-weight: bold;
    padding: 12px;
    width: 100%;
    border: none;
    border-radius: 5px;
    cursor: pointer;
    transition: background-color 0.3s ease;
    align-self: center;
    margin: auto;
    &:hover {
      background-color: #0056b3;
    }

    &:disabled {
      background-color: #ccc;
      cursor: not-allowed;
    }
  }

  &__footer {
    text-align: center;
    margin-top: 20px;
    display: flex;
    gap: 1rem;
    justify-content: center;
  }

  &__login-link {
    color: #007bff;
    text-decoration: none;

    &:hover {
      text-decoration: underline;
    }
  }

  .flex {
    display: flex;
    gap: 10px;
  }
}
