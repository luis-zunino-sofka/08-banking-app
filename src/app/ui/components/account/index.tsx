import { useCreateBankAccount } from "@core/hooks";
import { decryptAES } from "@core/utils";
import React from "react";
import { Toaster } from "react-hot-toast";
import { Button } from "../button";
import "./styles.scss";

export const CreateBankAccount: React.FC = () => {
  const { isLoading, accounts, errors, handleSubmit, register, onSubmit } =
    useCreateBankAccount();

  return (
    <div>
      <form onSubmit={handleSubmit(onSubmit)} className="form">
        <div style={{ display: "flex", gap: "10px" }}>
          <div>
            <div>
              <label htmlFor="accountType">Tipo de Cuenta</label>
              <select
                id="accountType"
                {...register("accountType", {
                  required: "Selecciona un tipo de cuenta",
                })}
              >
                <option key={1} value="">
                  Selecciona el tipo de cuenta
                </option>
                <option key={2} value="corriente">
                  Corriente
                </option>
              </select>
              <span className="error-message">
                {errors.accountType && errors.accountType.message}
              </span>
            </div>
            <div>
              <label htmlFor="initialBalance">Saldo Inicial</label>
              <input
                type="number"
                id="initialBalance"
                placeholder="Saldo inicial en $"
                min={0}
                {...register("initialBalance", {
                  required: "El saldo inicial es obligatorio",
                  min: {
                    value: 0,
                    message: "El saldo inicial no puede ser negativo",
                  },
                })}
              />
              <span className="error-message">
                {errors.initialBalance && errors.initialBalance.message}
              </span>
            </div>
            <Button label="Crear Cuenta" isLoading={isLoading} />
          </div>
          <div className="user-accounts">
            <p>Cuentas del usuario:</p>
            {accounts.map((a) => (
              <div key={a.accountId} className="user-accounts__item">
                <div className="user-accounts__info">
                  <span>NÂº de cuenta:</span> {decryptAES(a.encryptedNumber)}
                </div>
                <div className="user-accounts__info amount">
                  <span>Monto:</span> $ {a.amount}
                </div>
              </div>
            ))}
          </div>
        </div>
      </form>
      <Toaster />
    </div>
  );
};
