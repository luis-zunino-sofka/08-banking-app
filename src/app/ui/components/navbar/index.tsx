import "./styles.scss";
import { INavItem, IState } from "@core/interfaces";
import { FC } from "react";

export interface IDashboardProps {
  state: IState;
  isRefechingAccounts: boolean;
  navItems: INavItem[];
  refetchAccounts: () => void;
  handleLogout: () => void;
}

export const Navbar: FC<IDashboardProps> = (props) => {
  const {
    state,
    navItems,
    isRefechingAccounts,
    refetchAccounts,
    handleLogout,
  } = props;

  return (
    <nav className="navbar">
      <div className="navbar__title">
        <h1>Bienvenido {state.loginData?.username}</h1>
      </div>
      <div className="navbar__balance">
        <span>Saldo disponible: </span>
        <strong>${state.balance}</strong>
        <button
          disabled={isRefechingAccounts}
          className="navbar__reload"
          onClick={() => refetchAccounts()}
        >
          Actualizar
        </button>
      </div>
      <div className="navbar__nav">
        <ul className="navbar__list">
          {navItems.map((item) => (
            <li key={item.path} className="navbar-item">
              <button
                className={`navbar__link ${
                  item.isActive ? "navbar__link--active" : ""
                }`}
                onClick={item.onClick}
              >
                {item.label}
              </button>
            </li>
          ))}
          <li className="navbar__item">
            <button
              className="navbar__link navbar__logout"
              onClick={handleLogout}
            >
              Cerrar sesi√≥n
            </button>
          </li>
        </ul>
      </div>
    </nav>
  );
};
