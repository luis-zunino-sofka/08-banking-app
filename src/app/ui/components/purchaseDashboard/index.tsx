import { Toaster } from "react-hot-toast";
import { usePurchase } from "@core/hooks";
import { decryptAES } from "@core/utils";
import { Button } from "../button";
import "../../styles/styles.scss";

export const PurchaseDashboard = () => {
  const { isLoading, accounts, errors, register, handleSubmit, onSubmit } =
    usePurchase();

  return (
    <div>
      <form
        id="purchase-form"
        className="form"
        onSubmit={handleSubmit(onSubmit)}
      >
        <div>
          <label htmlFor="purchase-type">Selecciona el tipo de compra:</label>
          <select
            id="purchase-type"
            {...register("purchaseType", {
              required: "Por favor selecciona un tipo de compra",
            })}
          >
            <option value="">Elige el tipo de compra</option>
            <option value="physical">Establecimiento físico (Sin costo)</option>
            <option value="online">En línea (Costo de seguro de $5 USD)</option>
          </select>
          {errors.purchaseType && (
            <span className="error-message">{errors.purchaseType.message}</span>
          )}
        </div>
        <div>
          <label htmlFor="account-number-to-deposit">
            Selecciona la cuenta a depositar:
          </label>
          <select
            id="account-number-to-deposit"
            {...register("accountId", {
              required: "La cuenta bancaria a depositar es obligatorio",
            })}
          >
            <option value="">Elige una cuenta</option>
            {accounts.map((a) => (
              <option key={a.accountId} value={a.accountId}>
                Cuenta: {decryptAES(a.encryptedNumber)} - Monto: {a.amount}
              </option>
            ))}
          </select>

          {errors.accountId && (
            <span className="error-message">{errors.accountId.message}</span>
          )}
        </div>
        <div>
          <label htmlFor="purchase-amount">
            Ingresa el monto a depositar:
            <input
              type="number"
              id="purchase-amount"
              placeholder="Monto de la compra en $"
              {...register("purchaseAmount", {
                required: "El monto es obligatorio",
                min: {
                  value: 1,
                  message: "El monto debe ser mayor que $0",
                },
              })}
            />
          </label>
          {errors.purchaseAmount && (
            <span className="error-message">
              {errors.purchaseAmount.message}
            </span>
          )}
        </div>
        <Button label="Comprar" isLoading={isLoading} />
      </form>

      <Toaster />
    </div>
  );
};
