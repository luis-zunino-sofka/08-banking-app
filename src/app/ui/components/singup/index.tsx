import { useSignup } from "@core/hooks/signup/useSignup";
import { ROUTE_PATH } from "@core/constants";
import { Toaster } from "react-hot-toast";
import "./styles.scss";
import { InputSignup } from "./InputSignup";

export const SignupForm = () => {
  const { errors, isLoading, onSubmit, register, handleSubmit, watch } =
    useSignup();

  return (
    <div className="signup-form">
      <form
        id="register-form"
        className="signup-form__form"
        onSubmit={handleSubmit(onSubmit)}
      >
        <h2 className="signup-form__title">Regístrate</h2>
        <div className="flex">
          <InputSignup
            register={register("name", {
              required: "El nombre es obligatorio",
            })}
            name="name"
            type="text"
            label="Nombre:"
            errors={errors}
            placeHolder="Ingrese su nombre"
          />

          <InputSignup
            register={register("lastName", {
              required: "El apellido es obligatorio",
            })}
            name="lastName"
            type="text"
            label="Apellido:"
            errors={errors}
            placeHolder="Ingrese su apellido"
          />
        </div>
        <div className="flex">
          <InputSignup
            register={register("dni", { required: "El DNI es obligatorio" })}
            name="dni"
            type="text"
            label="DNI:"
            errors={errors}
            placeHolder="Ingrese su dni"
          />

          <InputSignup
            register={register("username", {
              required: "El nombre de usuario es obligatorio",
            })}
            name="username"
            type="text"
            label="Nombre de usuario:"
            errors={errors}
            placeHolder="Ingrese su nombre de usuario"
          />
        </div>
        <div className="flex">
          <InputSignup
            register={register("password", {
              required: "El nombre de usuario es obligatorio",
            })}
            name="password"
            type="password"
            label="Contraseña:"
            errors={errors}
            placeHolder="Contraseña"
          />

          <InputSignup
            register={register("confirmPassword", {
              required: "La confirmación de la contraseña es obligatoria",
              validate: (value) =>
                value === watch("password") || "Las contraseñas no coinciden",
            })}
            name="confirmPassword"
            type="password"
            label="Confirmar contraseña:"
            errors={errors}
            placeHolder="Confirmar contraseña"
          />
        </div>
        <button
          type="submit"
          className="signup-form__submit"
          disabled={isLoading}
        >
          Registrarse
        </button>
        <div className="signup-form__footer">
          <p>
            ¿Ya tienes cuenta?
            <a
              href={ROUTE_PATH.LOGIN}
              id="login-link"
              className="signup-form__login-link"
            >
              Inicia sesión aquí
            </a>
          </p>
        </div>
      </form>
      <Toaster />
    </div>
  );
};
