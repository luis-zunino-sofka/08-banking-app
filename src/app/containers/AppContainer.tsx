import { useNavbar } from "@core/hooks";
import { useAccountContext, useAppContext } from "@core/state";
import { Navbar } from "@ui/components/navbar";
import { AppLayout } from "@ui/layouts";
import React from "react";
import { Toaster } from "react-hot-toast";

export const AppContainer = ({ children }: { children: React.ReactNode }) => {
  const { state } = useAppContext();
  const { isRefechingAccounts, refetchAccounts } = useAccountContext();
  const { navItems, handleLogout } = useNavbar();

  return (
    <AppLayout>
      <Navbar
        state={state}
        handleLogout={handleLogout}
        refetchAccounts={refetchAccounts}
        navItems={navItems}
        isRefechingAccounts={isRefechingAccounts}
      />
      {children}
      <Toaster />
    </AppLayout>
  );
};
