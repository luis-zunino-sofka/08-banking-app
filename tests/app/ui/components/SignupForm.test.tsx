import {
  render,
  screen,
  fireEvent,
  waitFor,
  act,
} from "@testing-library/react";
import { SignupForm } from "../../../../src/app/ui/components/singup";
import { beforeEach, describe, expect, Mock, test, vi } from "vitest";
import React from "react";
import "@testing-library/jest-dom";
import { mockAccountState, mockAppState } from "../../../mocks/dataMocked";
import { AccountContextMock } from "../containers/AppContainer.test";
import { FormProvider, useForm } from "react-hook-form";

vi.mock("react-hot-toast");

export function renderWithReactHookForm(ui, { defaultValues = {} } = {}) {
  const Wrapper = ({ children }) => {
    const methods = useForm({ defaultValues });
    return <FormProvider {...methods}>{children}</FormProvider>;
  };

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}

describe("SignupForm", () => {
  let mockOnSubmit: Mock;
  let mockHandleSubmit = vi.fn();

  beforeEach(() => {
    mockOnSubmit = vi.fn((data) => console.log(data));
    mockHandleSubmit = vi.fn((fn) => fn(mockOnSubmit));

    renderWithReactHookForm(
      <AccountContextMock
        mockAppState={mockAppState}
        children={<SignupForm />}
        mockAccountState={mockAccountState}
      />
    );
  });

  test("Match snapshoot", () => {
    const { asFragment } = render(
      <AccountContextMock
        mockAppState={mockAppState}
        children={<SignupForm />}
        mockAccountState={mockAccountState}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("debe renderear el formulario correctamente", () => {
    screen.getByText(/Nombre:/i);
    screen.getByText(/Apellido:/i);
    screen.getByText(/DNI:/i);
    screen.getByText(/Nombre de usuario:/i);
    screen.getByText(/Confirmar contraseña:/i);
    screen.getByText(/^Contraseña:$/i);
    screen.getByText(/Registrarse/i);
    screen.getByText(/¿Ya tienes cuenta?/i);
  });

  test("debe mostrar validaciones si no hay campos completos", async () => {
    const submitButton = screen.getByText("Registrarse");
    await waitFor(() => fireEvent.click(submitButton));
    const nombre = screen.getByPlaceholderText(/^Ingrese su nombre$/i);
    const apellido = screen.getByPlaceholderText(/^Ingrese su apellido$/i);
    const dni = screen.getByPlaceholderText(/Ingrese su dni/i);
    const username = screen.getByPlaceholderText(
      /Ingrese su nombre de usuario/i
    );
    const password = screen.getByPlaceholderText(/^Contraseña$/i);
    const confirmPassword =
      screen.getByPlaceholderText(/Confirmar contraseña/i);

    expect(nombre).toHaveValue("");
    expect(apellido).toHaveValue("");
    expect(dni).toHaveValue("");
    expect(username).toHaveValue("");
    expect(password).toHaveValue("");
    expect(confirmPassword).toHaveValue("");

    screen.findAllByText("El nombre es obligatorio");
    screen.findAllByText("El apellido es obligatorio");
    screen.findAllByText("El DNI es obligatorio");
    screen.findAllByText("El nombre de usuario es obligatorio");
    screen.findAllByText("La contraseña es obligatoria");
    screen.findAllByText("La confirmación de la contraseña es obligatoria");

    expect(mockOnSubmit).not.toHaveBeenCalled();
  });

  test("puede llamar al onsubmit cuando esta validado", async () => {
    const nombre = screen.getByPlaceholderText(/^Ingrese su nombre$/i);
    const apellido = screen.getByPlaceholderText(/^Ingrese su apellido$/i);
    const dni = screen.getByPlaceholderText(/Ingrese su dni/i);
    const username = screen.getByPlaceholderText(
      /Ingrese su nombre de usuario/i
    );
    const password = screen.getByPlaceholderText(/^Contraseña$/i);
    const confirmPassword =
      screen.getByPlaceholderText(/Confirmar contraseña/i);
    waitFor(() => {
      fireEvent.change(nombre, {
        target: { value: "Pepe" },
      });
      fireEvent.change(apellido, {
        target: { value: "Pérez" },
      });
      fireEvent.change(dni, {
        target: { value: "12345678" },
      });
      fireEvent.change(username, {
        target: { value: "pepeperez" },
      });
      fireEvent.change(password, {
        target: { value: "123456" },
      });
      fireEvent.change(confirmPassword, {
        target: { value: "123456" },
      });
    });
    expect(nombre).toHaveValue("Pepe");
    expect(apellido).toHaveValue("Pérez");
    expect(dni).toHaveValue("12345678");
    expect(username).toHaveValue("pepeperez");
    expect(password).toHaveValue("123456");
    expect(confirmPassword).toHaveValue("123456");
    const submitButton = screen.getByText("Registrarse");
    waitFor(() => fireEvent.click(submitButton));

    waitFor(() => expect(submitButton).toBeDisabled());

    waitFor(() => expect(mockHandleSubmit).toHaveBeenCalled());
  });

  test("debe mostrar errores si las contraseñas no coinciden", async () => {
    act(() => {
      fireEvent.input(screen.getByPlaceholderText(/^Contraseña$/i), {
        target: { value: "123456" },
      });
      fireEvent.input(screen.getByPlaceholderText(/Confirmar contraseña/i), {
        target: { value: "654321" },
      });
    });
    const submitButton = screen.getByText("Registrarse");
    await waitFor(() => fireEvent.click(submitButton));

    await waitFor(() => screen.getByText("Las contraseñas no coinciden"));
  });

  test('debe redirigir a inicio de sesion cuando se hace clic en "Inicia sesión aquí"', () => {
    const registerLink = screen.getByRole("link", {
      name: /Inicia sesión aquí/i,
    });
    expect(registerLink).toHaveAttribute("href", "/");
  });
});
