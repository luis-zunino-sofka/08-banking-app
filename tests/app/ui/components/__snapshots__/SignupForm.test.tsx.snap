// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AccountProvider > Match snapshoot 1`] = `<DocumentFragment />`;

exports[`SignupForm > Match snapshoot 1`] = `
<DocumentFragment>
  <div
    class="signup-form"
  >
    <form
      class="signup-form__form"
      id="register-form"
    >
      <h2
        class="signup-form__title"
      >
        Regístrate
      </h2>
      <div
        class="flex"
      >
        <div
          class="signup-form__input-group"
        >
          <label
            class="signup-form__label"
            for="input-name"
          >
            Nombre:
          </label>
          <input
            class="signup-form__input"
            id="input-name"
            name="name"
            placeholder="Ingrese su nombre"
            type="text"
          />
        </div>
        <div
          class="signup-form__input-group"
        >
          <label
            class="signup-form__label"
            for="input-lastName"
          >
            Apellido:
          </label>
          <input
            class="signup-form__input"
            id="input-lastName"
            name="lastName"
            placeholder="Ingrese su apellido"
            type="text"
          />
        </div>
      </div>
      <div
        class="flex"
      >
        <div
          class="signup-form__input-group"
        >
          <label
            class="signup-form__label"
            for="input-dni"
          >
            DNI:
          </label>
          <input
            class="signup-form__input"
            id="input-dni"
            name="dni"
            placeholder="Ingrese su dni"
            type="text"
          />
        </div>
        <div
          class="signup-form__input-group"
        >
          <label
            class="signup-form__label"
            for="input-username"
          >
            Nombre de usuario:
          </label>
          <input
            class="signup-form__input"
            id="input-username"
            name="username"
            placeholder="Ingrese su nombre de usuario"
            type="text"
          />
        </div>
      </div>
      <div
        class="flex"
      >
        <div
          class="signup-form__input-group"
        >
          <label
            class="signup-form__label"
            for="input-password"
          >
            Contraseña:
          </label>
          <input
            class="signup-form__input"
            id="input-password"
            name="password"
            placeholder="Contraseña"
            type="password"
          />
        </div>
        <div
          class="signup-form__input-group"
        >
          <label
            class="signup-form__label"
            for="input-confirmPassword"
          >
            Confirmar contraseña:
          </label>
          <input
            class="signup-form__input"
            id="input-confirmPassword"
            name="confirmPassword"
            placeholder="Confirmar contraseña"
            type="password"
          />
        </div>
      </div>
      <button
        class="signup-form__submit"
        type="submit"
      >
        Registrarse
      </button>
      <div
        class="signup-form__footer"
      >
        <p>
          ¿Ya tienes cuenta?
        </p>
        <a
          class="signup-form__login-link"
          href="/"
          id="login-link"
        >
          Inicia sesión aquí
        </a>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
